### PR TITLE
<applet> is hardly a thing now

### PR DESCRIPTION
**Pull request by annevk**
document.applets (which always return an empty collection) and a mention in the HTML parser, but that's it.

This is for https://github.com/whatwg/html/pull/1399 and https://bugzilla.mozilla.org/show_bug.cgi?id=1386008.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260130551)

# Firefox (nightly) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260130552)

# Sauce (safari) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260130561)

# Chrome (unstable) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/260130562)

# Sauce (MicrosoftEdge) #

**Review by domenic — COMMENTED** [review]
It seems like there could be some useful negative tests created here. E.g. that there is no CSP violation generated; that applet doesn't show up in the window named properties (by name or ID); that they don't have priority over divs in the document.* properties.

I think we should also introduce a -historical.html test for everything in the OP of https://github.com/whatwg/html/pull/1399, perhaps consolidated into one file. Including user agent stylesheet stuff.

I know this is more work, but I do think especially given all the extra places we removed it from, having a nice set of tests for implementers will help.

**Review by zcorpan — COMMENTED** [review]
Changes LGTM but I agree with @domenic that we should have more negative tests

**Comment by annevk** [comment]
Good call, found a bug in Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=757434.

**Review by zcorpan — APPROVED** [review]

File: html/dom/historical.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+test(() => {
+  assert_equals(document.war, undefined);
+}, "document cannot find applet")
+
+test(() => {
+  assert_equals(self.war, undefined);
+}, "window cannot find applet")
+
+test(() => {
+  assert_equals(self.getComputedStyle(document.getElementsByTagName("applet")[0], "").cssFloat, "none");
```
Comment (domenic) [inline html/dom/historical.html]: Might be nice to test vertical-align too.

**Review by domenic — APPROVED** [review]
LGTM, although *ideally* we'd also have negative tests for the CSP violation stuff too... might be worth filing a tracking issue for that, at least.

**Comment by gsnedders** [comment]
Is there any reason this isn't merged?

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2017-08-21 15:10:27
Finished: 2017-08-21 15:21:00





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17629)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/266806874)

**Comment by annevk** [comment]
I'm not sure.

**Comment by domenic** [comment]
I mean, there's still some outstanding missing coverage per my last review. I guess I'll file an issue.

**Comment by annevk** [comment]
Ah yeah, that's probably why. But it seems better to have this checked in.

**Pull request merged**